python/pfs/drp/stella/lsf.py

```diff
@@ -0,0 +1,688 @@
+        """
+        if point is None:
+            point = self.getAveragePosition()
+        kernel = self.computeKernel(point.getX())
+        intRadius = int(radius)
+
+        flux = trapz([kernel[ii] for ii in range(-intRadius, intRadius + 1)])
+        if radius != intRadius:
+            residual = radius - intRadius
+            left, right = kernel.interpolate([-radius, radius])
+            flux += 0.5*residual*(kernel[-intRadius - 1] + kernel[-intRadius])
+            flux += 0.5*residual*(kernel[intRadius + 1] + kernel[intRadius])
+        return flux
+
+    def getLocalKernel(self, point=None):
+        """Return a 2D ``FixedKernel`` representing the Lsf at a point
+
+        Parameters
+        ----------
+        point : `lsst.geom.Point2D`, optional
+            Position at which to evaluate the LSF.
+
+        Returns
+        -------
+        kernel : `lsst.afw.math.FixedKernel`
+            Kernel representing the Lsf at the nominated point.
+        """
+        if point is None:
+            point = self.getAveragePosition()
+        kernel = self.computeKernel(point.getX())
+        image = ImageD(kernel.values[np.newaxis, :], True, Point2I(-kernel.center, 0))
+        return FixedKernel(image)
+
+    def getAveragePosition(self):
+        """Return the average position for which the Lsf is defined
+
+        Returns
+        -------
+        position : `lsst.geom.Point2D`, optional
+            Average position for which the Lsf is defined.
+        """
+        return Point2D(0.5*self.length, 0)
+
+    def computeBBox(self, point=None):
+        """Return the bounding box of the kernel image
+
+        Parameters
+        ----------
+        point : `lsst.geom.Point2D`, optional
+            Position at which to evaluate the LSF.
+
+        Returns
+        -------
+        bbox : `lsst.afw.geom.Box2I`
+            Bounding box for the kernel image.
+        """
+        if point is None:
+            point = self.getAveragePosition()
+        return self.computeKernelImage(point.getX()).getBBox()
+
+    @classmethod
+    def readFits(cls, filename):
+        """Read from file
+
+        Though the method name says "FITS", we read a pickle file, because
+        pickling takes care of subclasses for us.
+
+        Parameters
+        ----------
+        filename : `str`
+            Filename from which to read.
+
+        Returns
+        -------
+        self : ``cls``
+            `Lsf` object read from file.
+        """
+        with open(filename, "rb") as fd:
+            return pickle.load(fd)
+
+    def writeFits(self, filename):
+        """Write to file
+
+        Though the method name says "FITS", we read a pickle file, because
+        pickling takes care of subclasses for us.
+
+        Parameters
+        ----------
+        filename : `str`
+            Filename to which to write.
+        """
+        with open(filename, "wb") as fd:
+            pickle.dump(self, fd)
+
+
+def gaussian(indices, sigma):
```

Comment:
Why expose this?  And please don't call the width parameter `sigma` -- that implies statistics, not the convenient but inaccurate approximation of a PSF as a Gaussian

Reply:
I wouldn't have said it's exposed since it's not included in `__all__`.